Extract link from RadioFrance Podcast RSS Flux
https://radiofrance-podcast.net/podcast09/rss_20108.xml

In [2]:
import feedparser

#create an instance
news_feed = feedparser.parse(r'https://radiofrance-podcast.net/podcast09/rss_20108.xml')

#Check properties of the flux
print(news_feed.feed.keys())

dict_keys(['title', 'title_detail', 'links', 'link', 'subtitle', 'subtitle_detail', 'language', 'rights', 'rights_detail', 'updated', 'updated_parsed', 'generator_detail', 'generator', 'image', 'authors', 'author', 'author_detail', 'tags', 'itunes_explicit', 'publisher_detail', 'summary', 'summary_detail', 'itunes_new-feed-url', 'pa_new-feed-url', 'podcastrf_originstation', 'googleplay_block'])


In [7]:
#Check link
print(news_feed.feed.podcastrf_originstation)

1


In [19]:
print(news_feed.entries[0].key)

{'title': 'Ramsès II', 'title_detail': {'type': 'text/plain', 'language': None, 'base': 'https://radiofrance-podcast.net/podcast09/rss_20108.xml', 'value': 'Ramsès II'}, 'links': [{'rel': 'alternate', 'type': 'text/html', 'href': 'https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/ramses-ii-8432538'}, {'length': '15317676', 'type': 'audio/mpeg', 'href': 'https://rf.proxycast.org/2b11debe-28c4-4883-88a8-eadd8f247d8e/20108-15.02.2023-ITEMA_23243409-2023F35387E0004-21.mp3', 'rel': 'enclosure'}], 'link': 'https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/ramses-ii-8432538', 'summary': 'durée : 00:15:55 - Les Odyssées - par : Laure Grandbesançon. - Pharaon célèbre et adulé, Ramsès II est réputé pour ses victoires militaires et la construction de nombreux temples et statues... Et ça, partout en Egypte\xa0! Découvre les incroyables aventures du pharaon le plus connu du monde\xa0!', 'summary_detail': {'type': 'text/html', 'language': None, 'base': 'https://radiofrance-podc

In [18]:
print(news_feed.entries[0].keys())

dict_keys(['title', 'title_detail', 'links', 'link', 'summary', 'summary_detail', 'authors', 'author', 'author_detail', 'tags', 'id', 'guidislink', 'published', 'published_parsed', 'podcastrf_businessreference', 'podcastrf_magnetothequeid', 'itunes_title', 'image', 'itunes_explicit', 'subtitle', 'subtitle_detail', 'content', 'itunes_duration', 'googleplay_block'])


Print key of each items in column

In [26]:
for key in news_feed.entries[0].keys():
    print(f"{key}")

title
title_detail
links
link
summary
summary_detail
authors
author
author_detail
tags
id
guidislink
published
published_parsed
podcastrf_businessreference
podcastrf_magnetothequeid
itunes_title
image
itunes_explicit
subtitle
subtitle_detail
content
itunes_duration
googleplay_block


Details what is inside to find the right url

In [32]:
for key,value in news_feed.entries[0].items():
    print(key)
    print(value)
    

title
Ramsès II
title_detail
{'type': 'text/plain', 'language': None, 'base': 'https://radiofrance-podcast.net/podcast09/rss_20108.xml', 'value': 'Ramsès II'}
links
[{'rel': 'alternate', 'type': 'text/html', 'href': 'https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/ramses-ii-8432538'}, {'length': '15317676', 'type': 'audio/mpeg', 'href': 'https://rf.proxycast.org/2b11debe-28c4-4883-88a8-eadd8f247d8e/20108-15.02.2023-ITEMA_23243409-2023F35387E0004-21.mp3', 'rel': 'enclosure'}]
link
https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/ramses-ii-8432538
summary
durée : 00:15:55 - Les Odyssées - par : Laure Grandbesançon. - Pharaon célèbre et adulé, Ramsès II est réputé pour ses victoires militaires et la construction de nombreux temples et statues... Et ça, partout en Egypte ! Découvre les incroyables aventures du pharaon le plus connu du monde !
summary_detail
{'type': 'text/html', 'language': None, 'base': 'https://radiofrance-podcast.net/podcast09/rss_20108.xml', 'v

Example : 
links
[{'rel': 'alternate', 'type': 'text/html', 'href': 'https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/ramses-ii-8432538'}, {'length': '15317676', 'type': 'audio/mpeg', 'href': 'https://rf.proxycast.org/2b11debe-28c4-4883-88a8-eadd8f247d8e/20108-15.02.2023-ITEMA_23243409-2023F35387E0004-21.mp3', 'rel': 'enclosure'}]

In [44]:
print(news_feed.entries[0]['links'][1]['href'])

https://rf.proxycast.org/2b11debe-28c4-4883-88a8-eadd8f247d8e/20108-15.02.2023-ITEMA_23243409-2023F35387E0004-21.mp3


In [46]:
print(news_feed.entries[0]['title'])

Ramsès II


In [3]:
results_link = []
for i,entry in enumerate(news_feed.entries):
    results_link.append(news_feed.entries[i]['title'])
    results_link.append(news_feed.entries[i]['links'][1]['href'])

In [72]:
print(len(results_link))

111


In [87]:
results_link[0:6]

['Ramsès II',
 'https://rf.proxycast.org/2b11debe-28c4-4883-88a8-eadd8f247d8e/20108-15.02.2023-ITEMA_23243409-2023F35387E0004-21.mp3',
 'Helen Keller',
 'https://rf.proxycast.org/fafa25c2-58d3-48e5-bfeb-5277d1e9fbea/20108-15.02.2023-ITEMA_23243409-2023F35387E0003-21.mp3',
 'La découverte de la grotte Cosquer',
 'https://rf.proxycast.org/b3b43433-b641-4171-821e-8020b19d8936/20108-15.02.2023-ITEMA_23243409-2023F35387E0002-21.mp3']

# Donwload

In [4]:
import requests
import os

In [5]:
os.getcwd()

'C:\\Users\\matin\\mp3'

Check that : in title is making an issue ?
eg. Rosalind Franklin  => title Rosalind Franklin : la femme qui a découvert le secret de l'ADN

REPLACE all not ok character in the title string.

In [11]:
print('start')
for i in range(0,112,2):
    name = results_link[i]
    url = results_link[i+1]
    r = requests.get(url, allow_redirects=True)
    
    #make the file name valid for windows user removing not alphanum character except "dot" and "_" underscore
    validname = "".join( x for x in name if (x.isalnum() or x in "._- "))
    
    open(validname +'.mp3', 'wb').write(r.content)
    print(f'downloaded item {i}')
    
print('end')
    

start
downloaded item 0
downloaded item 2
downloaded item 4
downloaded item 6
downloaded item 8
downloaded item 10
downloaded item 12
downloaded item 14
downloaded item 16
downloaded item 18
downloaded item 20
downloaded item 22
downloaded item 24
downloaded item 26
downloaded item 28
downloaded item 30
downloaded item 32
downloaded item 34
downloaded item 36
downloaded item 38
downloaded item 40
downloaded item 42
downloaded item 44
downloaded item 46
downloaded item 48
downloaded item 50
downloaded item 52
downloaded item 54
downloaded item 56
downloaded item 58
downloaded item 60
downloaded item 62
downloaded item 64
downloaded item 66
downloaded item 68
downloaded item 70
downloaded item 72
downloaded item 74
downloaded item 76
downloaded item 78
downloaded item 80
downloaded item 82
downloaded item 84
downloaded item 86
downloaded item 88
downloaded item 90
downloaded item 92
downloaded item 94
downloaded item 96
downloaded item 98
downloaded item 100
downloaded item 102
downloade

In [8]:
for i in range(6,20,2):
    print(i)

6
8
10
12
14
16
18


In [11]:
A = 'test'
pre, ext = os.path.splitext(A)
B = os.rename(pre,'test' + ".mp3")


Make the name a valide windows filenmae

In [8]:
print('start')
for i in range(0,130,2):
    name = results_link[i]
    validname = "".join( x for x in name if (x.isalnum() or x in "._- "))
    print(validname)
    
print('end')


start
Ramsès II
Helen Keller
La découverte de la grotte Cosquer
Ching Shih la pirate chinoise
Ramsès II raconté par Laure Grandbesançon
Henri Guillaumet  héros de laéropostale
Joséphine Baker
À la recherche du monstre des profondeurs
Elliot Ness contre Al Capone
Vercingétorix contre César
Rosalind Franklin la femme qui a découvert le secret de lADN
Percy Fawcett et La mystérieuse cité de Z épisode 2
Percy Fawcett et la mystérieuse cité de Z Episode 1
Aladin et le sorcier maléfique
Les 1001 nuits 44  Les trois princes amoureux
Les 1001 nuits 34  Les  incroyables aventures de Sindbad le marin
Aladin et le sorcier maléfique
Les 1001 nuits 14  Shéhérazade conteuse de génie
Yasuké le premier samouraï noir de lHistoire
Laskarina Bouboulina héroïne de la résistance grecque
Qui est Ötzi la momie des glaces
Les Odyssées du Louvre 1515  Les Odyssées du Louvre Enquête autour de la fascinante Vénus de Milo
Les Odyssées du Louvre 1415  Les Odyssées du Louvre  Eugène Delacroix grand peintre romantiq